docs/reference/viewsets.md

```diff
@@ -0,0 +1,62 @@
+# Viewsets
+
+Viewsets are Wagtail's mechanism for defining a group of related admin views with shared properties, as a single unit. See [Generic views](../extending/generic_views).
+
+```{eval-rst}
+
+ViewSet
+-------
+
+.. autoclass:: wagtail.admin.viewsets.base.ViewSet
+
+   .. automethod:: on_register
+   .. automethod:: get_urlpatterns
+   .. automethod:: get_url_name
+
+ModelViewSet
+------------
```

Comment:
Would it be possible to split out each code autoclass into its own eval-rst block and then put the headers in markdown?